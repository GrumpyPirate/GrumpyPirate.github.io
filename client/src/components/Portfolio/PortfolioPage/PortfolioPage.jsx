// React
import React, { PureComponent } from 'react'
import PropTypes from 'prop-types'

import { EventEmitter } from 'fbemitter'

// Vendor
import marked from 'marked'

// Styles
import './PortfolioPage.scss'

// Components
import PortfolioPageHeader from 'components/Portfolio/PortfolioPageHeader/PortfolioPageHeader.jsx'
import PortfolioPageContent from 'components/Portfolio/PortfolioPageContent/PortfolioPageContent.jsx'
import PortfolioCarousel from 'components/Portfolio/PortfolioCarousel/PortfolioCarousel.jsx'
import PortfolioDeviceLineup from 'components/Portfolio/PortfolioDeviceLineup/PortfolioDeviceLineup.jsx'

// Services
import ContentService from 'services/ContentService'

class PortfolioPage extends PureComponent {
  constructor (props) {
    super(props)

    // Service instances
    this.contentService = new ContentService()

    this.state = {
      portfolioItem: null
    } // /state
  } // /constructor

  getDescription (markdownDescription) {
    const markedOptions = { sanitize: true }
    return marked(this.state.portfolioItem.description, markedOptions)
  } // /getDescription

  getPortfolioItem (slug) {
    return this.contentService.getSinglePortfolioItem(slug)
  } // /getPortfolioItem

  // Lifecycle
  componentWillMount () {
    this.props.emitter.emit('startLoading')

    this.getPortfolioItem(this.props.match.params.portfolioItemSlug)
      .then(item => {
        this.setState({
          portfolioItem: item
        })

        this.props.emitter.emit('stopLoading')
      })
      .catch((e) => {
        console.error(e)
        this.props.emitter.emit('stopLoading')
      })
  } // /componentWillMount

  render () {
    return (
      <div className="pf-page">
        {this.state.portfolioItem &&
          <div>
            <PortfolioPageHeader
              bgImage={this.state.portfolioItem.headerImgSrc}
              title={this.state.portfolioItem.title}
              tech={this.state.portfolioItem.tech}
            />

            {/* Content */}
            <PortfolioPageContent>
              <div className="container-fluid">
                <div className="row justify-content-center align-items-center">
                  <div className="col-12 col-sm-10">
                    {this.state.portfolioItem.supportingImageSrc
                      ? (
                        <div className="row align-items-lg-center">
                          <div
                            className="col-12 col-md col-xl-6"
                            dangerouslySetInnerHTML={{
                              __html: this.getDescription(this.state.portfolioItem.description)
                            }}
                          />

                          <div className="col-12 col-md-6 col-xl-5 push-xl-1">
                            {/* Supporting image 1 */}
                            <hr className="mt-1 hidden-md-up"/>
                            <figure className="mb-0">
                              <img src={this.state.portfolioItem.supportingImageSrc} alt="" className="pf-page__supporting-image w-100" />
                            </figure>
                          </div>
                        </div>
                      )
                      : (
                        <div dangerouslySetInnerHTML={{
                          __html: this.getDescription(this.state.portfolioItem.description)}}
                        />
                      )
                    }

                    <hr/>

                    {/* Device Carousel, for mobile (sm-) only */}
                    <div className="hidden-md-up">
                      <PortfolioCarousel
                        desktop={this.state.portfolioItem.previews.desktop}
                        tablet={this.state.portfolioItem.previews.tablet}
                        mobile={this.state.portfolioItem.previews.mobile} />
                    </div>

                    {/* Device Lineup, for tablet up */}
                    <div className="hidden-sm-down">
                      <PortfolioDeviceLineup
                        desktop={this.state.portfolioItem.previews.desktop}
                        tablet={this.state.portfolioItem.previews.tablet}
                        mobile={this.state.portfolioItem.previews.mobile} />
                    </div>

                    <hr/>

                    <div className="text-center mt-2 mt-md-4">
                      <a href={this.state.portfolioItem.url} className="btn btn-secondary" target="_blank">Visit {this.state.portfolioItem.title}</a>
                    </div>
                  </div>
                </div>
              </div>
            </PortfolioPageContent>
          </div>
        }
      </div>
    )
  } // /render()
} // /class PortfolioPage extends Component

PortfolioPage.propTypes = {
  emitter: PropTypes.instanceOf(EventEmitter).isRequired,
  match: PropTypes.shape({
    params: PropTypes.shape({
      portfolioItemSlug: PropTypes.string.isRequired
    })
  })
}

export default PortfolioPage
