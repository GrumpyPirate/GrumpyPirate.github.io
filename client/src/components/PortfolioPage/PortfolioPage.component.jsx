import React, { Component } from 'react';
import { Redirect } from 'react-router-dom';
import PropTypes from 'prop-types';
import ReactMarkdown from 'react-markdown';
import isEmpty from 'lodash-es/isEmpty';

import Container from 'components/Layout/Container/Container.component';
import Row from 'components/Layout/Row/Row.component';
import Column from 'components/Layout/Column/Column.component';
import Button from 'components/Layout/Button/Button.component';
import Page from 'components/Page/Page.component';
import PortfolioPageHeader from 'components/PortfolioPageHeader/PortfolioPageHeader.component';
import PortfolioPageContent from 'components/PortfolioPageContent/PortfolioPageContent.component';
import PortfolioCarousel from 'components/PortfolioCarousel/PortfolioCarousel.component';
import PortfolioDeviceLineup from 'components/PortfolioDeviceLineup/PortfolioDeviceLineup.component';

import classes from './PortfolioPage.component.scss';

class PortfolioPage extends Component {
  componentDidMount() {
    this.fetchPortfolioItems();
  }

  componentDidUpdate() {
    const { hasFetched, portfolioItem } = this.props;

    if (hasFetched && isEmpty(portfolioItem)) {
      return <Redirect to="/404" />;
    }
  }

  fetchPortfolioItems() {
    const { portfolioItem, hasFetched, requestPortfolioItems } = this.props;
    const shouldFetchItems = Boolean(isEmpty(portfolioItem) && !hasFetched);

    if (shouldFetchItems) {
      return requestPortfolioItems();
    }
  }

  render() {
    const { portfolioItem } = this.props;
    if (!portfolioItem) return null;

    return (
      <Page>
        <div className={classes['portfolio-page']}>
          <PortfolioPageHeader
            bgImage={portfolioItem.headerImgSrc}
            title={portfolioItem.title}
            tech={portfolioItem.tech}
          />

          <PortfolioPageContent>
            <Container>
              <Row justify="center">
                <Column spanSM={10}>
                  {portfolioItem.supportingImageSrc
                    ? (
                      <Row alignLG="middle">
                        <Column span={12} spanMD="auto" spanXL={6}>
                          <ReactMarkdown source={portfolioItem.description} />
                        </Column>

                        <Column span={12} spanMD={6} spanXL={5} pushXL={1}>
                          <figure className={classes['portfolio-page__supporting-image']}>
                            <img src={portfolioItem.supportingImageSrc} alt="" />
                          </figure>
                        </Column>
                      </Row>
                    )
                    : <ReactMarkdown source={portfolioItem.description} />
                  }

                  <hr className={classes['portfolio-page__content-divider']} />

                  <PortfolioCarousel
                    desktopImage={portfolioItem.previews.desktop}
                    tabletImage={portfolioItem.previews.tablet}
                    mobileImage={portfolioItem.previews.mobile}
                  />

                  <PortfolioDeviceLineup
                    desktopImage={portfolioItem.previews.desktop}
                    tabletImage={portfolioItem.previews.tablet}
                    mobileImage={portfolioItem.previews.mobile}
                  />

                  <hr className={classes['portfolio-page__content-divider']} />

                  <div className={classes['portfolio-page__external-link']}>
                    <Button
                      to={portfolioItem.url}
                      external
                      secondary
                    >
                      Visit
                      {' '}
                      {portfolioItem.title}
                    </Button>
                  </div>
                </Column>
              </Row>
            </Container>
          </PortfolioPageContent>
        </div>
      </Page>
    );
  }
}

PortfolioPage.propTypes = {
  portfolioItem: PropTypes.shape({
    description: PropTypes.string.isRequired,
    headerImgSrc: PropTypes.string.isRequired,
    title: PropTypes.string.isRequired,
    tech: PropTypes.arrayOf(
      PropTypes.shape({
        icon: PropTypes.string.isRequired,
        name: PropTypes.string.isRequired,
      }),
    ).isRequired,
    supportingImageSrc: PropTypes.string,
    url: PropTypes.string.isRequired,
    previews: PropTypes.shape({
      desktop: PropTypes.string.isRequired,
      tablet: PropTypes.string.isRequired,
      mobile: PropTypes.string.isRequired,
    }).isRequired,
  }),
  hasFetched: PropTypes.bool.isRequired,
  requestPortfolioItems: PropTypes.func.isRequired,
};

PortfolioPage.defaultProps = {
  portfolioItem: null,
};

export default PortfolioPage;
