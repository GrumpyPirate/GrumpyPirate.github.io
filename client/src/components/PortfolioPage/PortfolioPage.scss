@import 'core';

.portfolio-page {
  &__supporting-image {
    margin: 0;
    border-radius: .25rem;

    > img {
      display: block;
      width: 100%;
      height: auto;
    }
  }

  &__content-divider {
    border-style: solid none none;
    border-color: currentColor;
    border-width: 1px;
    margin: 2rem 0;
    opacity: .25;
  }

  &__external-link {
    text-align: center;
    margin-top: 2rem;

    @include respond-up(md) {
      margin-top: 4rem;
    }
  }
}
