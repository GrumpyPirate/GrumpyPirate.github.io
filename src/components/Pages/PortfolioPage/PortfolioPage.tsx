import { useQuery } from '@apollo/client';
import React, { FunctionComponent } from 'react';
import styled from 'styled-components';

import Page from 'components/Page/Page';
import PageContent from 'components/PageContent/PageContent';
import PageHeader from 'components/PageHeader/PageHeader';
import PageHeaderSubtitle from 'components/PageHeaderSubtitle/PageHeaderSubtitle';
import PortfolioList from 'components/PortfolioList/PortfolioList';
import Spinner from 'components/Spinner/Spinner';
import { GET_PORTFOLIO_ITEM_LIST, GetPortfolioItemsResponse } from 'queries';
import { PropsWithClassName } from 'types/common';

const PortfolioPage: FunctionComponent<PropsWithClassName> = function ({ className }) {
  const { loading } = useQuery<GetPortfolioItemsResponse>(GET_PORTFOLIO_ITEM_LIST);

  return (
    <Page className={className}>
      <PageHeader title="Portfolio">
        <PageHeaderSubtitle>Projects I&apos;ve worked on</PageHeaderSubtitle>
      </PageHeader>

      <PageContent>
        {loading && <Spinner />}
        <PortfolioList />
      </PageContent>
    </Page>
  );
};

export default styled(PortfolioPage)`
  ${PageContent} {
    padding-top: 0;
  }
`;
