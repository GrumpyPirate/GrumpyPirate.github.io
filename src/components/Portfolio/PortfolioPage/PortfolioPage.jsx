// React
import React, { PureComponent } from 'react'

// Vendor
import marked from 'marked'

// Styles
// import './PortfolioPage.scss'

// Components
import PortfolioPageHeader from 'components/Portfolio/PortfolioPageHeader/PortfolioPageHeader.jsx'
import PortfolioPageContent from 'components/Portfolio/PortfolioPageContent/PortfolioPageContent.jsx'
import PortfolioCarousel from 'components/Portfolio/PortfolioCarousel/PortfolioCarousel.jsx'
import PortfolioDeviceLineup from 'components/Portfolio/PortfolioDeviceLineup/PortfolioDeviceLineup.jsx'

// Services
import ContentService from 'services/ContentService'

class PortfolioPage extends PureComponent {
  constructor (props) {
    super(props)

    this.state = {
      contentService: new ContentService(),
      portfolioItem: null,
      portfolioItemSlug: this.props.match.params.portfolioItemSlug
    }
  }

  getDescription (markdownDescription) {
    const markedOptions = {
      sanitize: true
    }

    return marked(this.state.portfolioItem.description, markedOptions)
  }

  componentWillMount () {
    this.props.emitter.emit('startLoading')

    this.state.contentService
      .getSinglePortfolioItem(this.state.portfolioItemSlug)
      .then((item) => {
        this.setState({
          portfolioItem: item
        })

        this.props.emitter.emit('stopLoading')
      })
      .catch((e) => {
        console.error(e)
        this.props.emitter.emit('stopLoading')
      })
  } // /componentWillMount

  render () {
    return (
      <div className="pf-page">
        {this.state.portfolioItem &&
          <div>
            <PortfolioPageHeader bgImage={this.state.portfolioItem.headerImgSrc} title={this.state.portfolioItem.title} tech={this.state.portfolioItem.tech}/>

            {/* Content */}
            <PortfolioPageContent>
              <div className="container-fluid">
                <div className="row justify-content-center align-items-center">
                  <div className="col-12 col-sm-10">
                    {this.state.portfolioItem.supportingImageSrc
                      ? (
                        <div className="row align-items-lg-center">
                          <div className="col-12 col-md" dangerouslySetInnerHTML={{__html: this.getDescription(this.state.portfolioItem.description)}}></div>

                          <div className="col-12 col-md-6 col-xl-5">
                            {/* Supporting image 1 */}
                            <hr className="mt-1 hidden-md-up"/>
                            <figure className="mb-0">
                              <img src={this.state.portfolioItem.supportingImageSrc} alt="" className="w-100" />
                            </figure>
                          </div>
                        </div>
                      )
                      : (
                        <div>
                          <p>Lorem ipsum dolor sit amet, consectetur adipisicing elit. Quae enim quisquam exercitationem doloremque porro veritatis repellat ea, mollitia maxime, ullam aliquam! Deleniti, blanditiis nostrum.
                          </p>
                          <p>Lorem ipsum dolor sit amet, consectetur adipisicing elit. Animi, deserunt. Iste reiciendis magnam temporibus quidem, quis hic libero molestiae. Cum facere, recusandae reiciendis corrupti ipsa placeat quas explicabo rem debitis quod magnam sequi. Minus pariatur totam, maxime deleniti a ullam cum rem praesentium, minima reiciendis, ad quaerat repudiandae.
                          </p>
                        </div>
                      )
                    }

                    <hr/>

                    {/* Device Carousel, for mobile (sm-) only */}
                    <div className="hidden-md-up">
                      <PortfolioCarousel
                        desktop={this.state.portfolioItem.previews.desktop}
                        tablet={this.state.portfolioItem.previews.tablet}
                        mobile={this.state.portfolioItem.previews.mobile} />
                    </div>

                    {/* Device Lineup, for tablet up */}
                    <div className="hidden-sm-down">
                      <PortfolioDeviceLineup
                        desktop={this.state.portfolioItem.previews.desktop}
                        tablet={this.state.portfolioItem.previews.tablet}
                        mobile={this.state.portfolioItem.previews.mobile} />
                    </div>

                    <hr/>

                    <div className="text-center mt-2 mt-md-4">
                      <a href={this.state.portfolioItem.url} className="btn btn-secondary" target="_blank">Visit {this.state.portfolioItem.title}</a>
                    </div>
                  </div>
                </div>
              </div>
            </PortfolioPageContent>
          </div>
        }
      </div>
    )
  } // /render()
} // /class PortfolioPage extends Component

export default PortfolioPage
