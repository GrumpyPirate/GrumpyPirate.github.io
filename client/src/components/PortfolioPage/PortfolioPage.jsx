import React, { Component } from 'react';
import { withRouter, Redirect } from 'react-router-dom';
import PropTypes from 'prop-types';
import ReactMarkdown from 'react-markdown';
import isEmpty from 'lodash-es/isEmpty';

import Page from 'components/Page/Page';
import PortfolioPageHeader from 'components/PortfolioPageHeader/PortfolioPageHeader';
import PortfolioPageContent from 'components/PortfolioPageContent/PortfolioPageContent';
import PortfolioCarousel from 'components/PortfolioCarousel/PortfolioCarousel';
import PortfolioDeviceLineup from 'components/PortfolioDeviceLineup/PortfolioDeviceLineup';

import './PortfolioPage.scss';

class PortfolioPage extends Component {
  componentDidMount() {
    this.fetchPortfolioItems();
  }

  componentDidUpdate() {
    const { hasFetched, portfolioItem } = this.props;

    if (hasFetched && isEmpty(portfolioItem)) {
      return <Redirect to="/404" />;
    }
  }

  fetchPortfolioItems() {
    const { portfolioItem, hasFetched, requestPortfolioItems } = this.props;
    const shouldFetchItems = Boolean(isEmpty(portfolioItem) && !hasFetched);

    if (shouldFetchItems) {
      return requestPortfolioItems();
    }
  }

  render() {
    const { portfolioItem } = this.props;
    if (!portfolioItem) return null;

    return (
      <Page>
        <div className="pf-page">
          <PortfolioPageHeader
            bgImage={portfolioItem.headerImgSrc}
            title={portfolioItem.title}
            tech={portfolioItem.tech}
          />

          <PortfolioPageContent>
            <div className="container-fluid">
              <div className="row justify-content-center align-items-center">
                <div className="col-12 col-sm-10">
                  {portfolioItem.supportingImageSrc
                    ? (
                      <div className="row align-items-lg-center">
                        <div className="col-12 col-md col-xl-6">
                          <ReactMarkdown source={portfolioItem.description} />
                        </div>

                        <div className="col-12 col-md-6 col-xl-5 push-xl-1">
                          <hr className="mt-1 hidden-md-up" />
                          <figure className="mb-0">
                            <img
                              src={portfolioItem.supportingImageSrc}
                              alt=""
                              className="pf-page__supporting-image w-100"
                            />
                          </figure>
                        </div>
                      </div>
                    )
                    : <ReactMarkdown source={portfolioItem.description} />
                  }

                  <hr />

                  <div className="hidden-md-up">
                    <PortfolioCarousel
                      desktop={portfolioItem.previews.desktop}
                      tablet={portfolioItem.previews.tablet}
                      mobile={portfolioItem.previews.mobile}
                    />
                  </div>

                  <div className="hidden-sm-down">
                    <PortfolioDeviceLineup
                      desktop={portfolioItem.previews.desktop}
                      tablet={portfolioItem.previews.tablet}
                      mobile={portfolioItem.previews.mobile}
                    />
                  </div>

                  <hr />

                  <div className="text-center mt-2 mt-md-4">
                    <a
                      href={portfolioItem.url}
                      className="btn btn-secondary"
                      target="_blank"
                      rel="noreferrer noopener"
                    >
                      Visit
                      {' '}
                      {portfolioItem.title}
                    </a>
                  </div>
                </div>
              </div>
            </div>
          </PortfolioPageContent>
        </div>
      </Page>
    );
  }
}

PortfolioPage.propTypes = {
  portfolioItem: PropTypes.shape({
    description: PropTypes.string.isRequired,
    headerImgSrc: PropTypes.string.isRequired,
    title: PropTypes.string.isRequired,
    tech: PropTypes.arrayOf(PropTypes.shape({

    })).isRequired,
    supportingImageSrc: PropTypes.string,
    url: PropTypes.string.isRequired,
    previews: PropTypes.shape({
      desktop: PropTypes.string.isRequired,
      tablet: PropTypes.string.isRequired,
      mobile: PropTypes.string.isRequired,
    }).isRequired,
  }),
  hasFetched: PropTypes.bool.isRequired,
  history: PropTypes.shape({
    push: PropTypes.func.isRequired,
  }).isRequired,
  match: PropTypes.shape({
    params: PropTypes.shape({
      slug: PropTypes.string.isRequired,
    }).isRequired,
  }).isRequired,
  requestPortfolioItems: PropTypes.func.isRequired,
};

PortfolioPage.defaultProps = {
  portfolioItem: null,
};

export default withRouter(PortfolioPage);
