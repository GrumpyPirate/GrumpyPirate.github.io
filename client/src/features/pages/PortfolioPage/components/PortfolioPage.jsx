import React, { Component, Fragment } from 'react';
import { withRouter, Redirect } from 'react-router-dom';
import PropTypes from 'prop-types';
import ReactMarkdown from 'react-markdown';
import isEmpty from 'lodash/isEmpty';

import PortfolioPageHeader from 'components/Portfolio/PortfolioPageHeader/PortfolioPageHeader';
import PortfolioPageContent from 'components/Portfolio/PortfolioPageContent/PortfolioPageContent';
import PortfolioCarousel from 'components/Portfolio/PortfolioCarousel/PortfolioCarousel';
import PortfolioDeviceLineup from 'components/Portfolio/PortfolioDeviceLineup/PortfolioDeviceLineup';

import './PortfolioPage.scss';

class PortfolioPage extends Component {
  componentDidMount() {
    this.fetchPortfolioItems();
  }

  componentDidUpdate() {
    const { hasFetched, portfolioItem } = this.props;

    if (hasFetched && isEmpty(portfolioItem)) {
      return <Redirect to="/404" />;
    }
  }

  fetchPortfolioItems() {
    const { portfolioItem, hasFetched, requestPortfolioItems } = this.props;
    const shouldFetchItems = Boolean(isEmpty(portfolioItem) && !hasFetched);

    if (shouldFetchItems) {
      return requestPortfolioItems();
    }
  }

  render() {
    const { portfolioItem } = this.props;
    if (!portfolioItem) return null;

    return (
      <div className="pf-page">
        <Fragment>
          <PortfolioPageHeader
            bgImage={portfolioItem.headerImgSrc}
            title={portfolioItem.title}
            tech={portfolioItem.tech}
          />

          <PortfolioPageContent>
            <div className="container-fluid">
              <div className="row justify-content-center align-items-center">
                <div className="col-12 col-sm-10">
                  {portfolioItem.supportingImageSrc
                    ? (
                      <div className="row align-items-lg-center">
                        <div className="col-12 col-md col-xl-6">
                          <ReactMarkdown source={portfolioItem.description} />
                        </div>

                        <div className="col-12 col-md-6 col-xl-5 push-xl-1">
                          <hr className="mt-1 hidden-md-up" />
                          <figure className="mb-0">
                            <img
                              src={portfolioItem.supportingImageSrc}
                              alt=""
                              className="pf-page__supporting-image w-100"
                            />
                          </figure>
                        </div>
                      </div>
                    )
                    : <ReactMarkdown source={portfolioItem.description} />
                  }

                  <hr />

                  <div className="hidden-md-up">
                    <PortfolioCarousel
                      desktop={portfolioItem.previews.desktop}
                      tablet={portfolioItem.previews.tablet}
                      mobile={portfolioItem.previews.mobile}
                    />
                  </div>

                  <div className="hidden-sm-down">
                    <PortfolioDeviceLineup
                      desktop={portfolioItem.previews.desktop}
                      tablet={portfolioItem.previews.tablet}
                      mobile={portfolioItem.previews.mobile}
                    />
                  </div>

                  <hr />

                  <div className="text-center mt-2 mt-md-4">
                    <a
                      href={portfolioItem.url}
                      className="btn btn-secondary"
                      target="_blank"
                      rel="noreferrer noopener"
                    >
                      Visit
                      {' '}
                      {portfolioItem.title}
                    </a>
                  </div>
                </div>
              </div>
            </div>
          </PortfolioPageContent>
        </Fragment>
      </div>
    );
  }
}

PortfolioPage.propTypes = {
  portfolioItem: PropTypes.shape({
    description: PropTypes.string.isRequired,
    headerImgSrc: PropTypes.string.isRequired,
    title: PropTypes.string.isRequired,
    tech: PropTypes.arrayOf(PropTypes.shape({

    })).isRequired,
    supportingImageSrc: PropTypes.string,
    url: PropTypes.string.isRequired,
    previews: PropTypes.shape({
      desktop: PropTypes.string.isRequired,
      tablet: PropTypes.string.isRequired,
      mobile: PropTypes.string.isRequired,
    }).isRequired,
  }),
  hasFetched: PropTypes.bool.isRequired,
  history: PropTypes.shape({
    push: PropTypes.func.isRequired,
  }).isRequired,
  match: PropTypes.shape({
    params: PropTypes.shape({
      slug: PropTypes.string.isRequired,
    }).isRequired,
  }).isRequired,
  requestPortfolioItems: PropTypes.func.isRequired,
};

PortfolioPage.defaultProps = {
  portfolioItem: null,
};

export default withRouter(PortfolioPage);
