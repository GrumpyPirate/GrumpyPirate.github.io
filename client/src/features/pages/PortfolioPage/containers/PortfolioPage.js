import { connect } from 'react-redux';
import { withRouter } from 'react-router-dom';
import PropTypes from 'prop-types';

import { requestPortfolioItems } from 'actions/portfolio/portfolio';

import PortfolioPage from '../components/PortfolioPage';

const mapStateToProps = (state, ownProps) => ({
  portfolioItem: state.portfolio.portfolioItems
    .find(item => item.slug === ownProps.match.params.slug),
  hasFetched: state.portfolio.hasFetched,
});

const mapDispatchToProps = { requestPortfolioItems };

const PortfolioPageContainer = withRouter(connect(
  mapStateToProps,
  mapDispatchToProps,
)(PortfolioPage));

PortfolioPageContainer.propTypes = {
  match: PropTypes.shape({
    params: PropTypes.shape({
      slug: PropTypes.string.isRequired,
    }).isRequired,
  }).isRequired,
};

export default PortfolioPageContainer;
