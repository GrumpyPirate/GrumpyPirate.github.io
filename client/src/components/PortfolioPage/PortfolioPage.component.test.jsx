import React from 'react';
import { shallow } from 'enzyme';

import PortfolioPage from './PortfolioPage.component';

import fixture from './PortfolioPage.component.fixture';

describe('Components', () => {
  describe('PortfolioPage', () => {
    const props = {
      ...fixture,
      history: {
        push: jest.fn(),
      },
      requestPortfolioItems: jest.fn(),
    };

    it('should be defined', () => {
      expect(PortfolioPage).toBeDefined();
    });

    it('should be a function', () => {
      expect(typeof PortfolioPage).toBe('function');
    });

    it('should render as expected, without crashing', () => {
      const wrapper = shallow(<PortfolioPage {...props} />);

      expect(wrapper).toMatchSnapshot();
    });
  });
});
