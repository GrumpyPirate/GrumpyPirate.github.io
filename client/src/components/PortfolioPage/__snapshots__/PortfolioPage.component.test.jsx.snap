// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components PortfolioPage should render as expected, without crashing 1`] = `
<withRouter(Page)>
  <div
    className="portfolio-page"
  >
    <PortfolioPageHeader
      bgImage="https://devotement.com/anticorn/pajahuello?a=rhynchophoran&b=toyer#prescience"
      tech={
        Array [
          Object {
            "icon": "Glycolaldehyde",
            "name": "youngling",
          },
        ]
      }
      title="Scrupulum Misphrase"
    />
    <PortfolioPageContent>
      <Container>
        <Row
          align={null}
          alignLG={null}
          alignMD={null}
          alignSM={null}
          alignXL={null}
          justify="center"
          justifyLG={null}
          justifyMD={null}
          justifySM={null}
          justifyXL={null}
        >
          <Column
            pull={null}
            pullLG={null}
            pullMD={null}
            pullSM={null}
            pullXL={null}
            push={null}
            pushLG={null}
            pushMD={null}
            pushSM={null}
            pushXL={null}
            span={null}
            spanLG={null}
            spanMD={null}
            spanSM={10}
            spanXL={null}
          >
            <Row
              align={null}
              alignLG="middle"
              alignMD={null}
              alignSM={null}
              alignXL={null}
              justify={null}
              justifyLG={null}
              justifyMD={null}
              justifySM={null}
              justifyXL={null}
            >
              <Column
                pull={null}
                pullLG={null}
                pullMD={null}
                pullSM={null}
                pullXL={null}
                push={null}
                pushLG={null}
                pushMD={null}
                pushSM={null}
                pushXL={null}
                span={12}
                spanLG={null}
                spanMD="auto"
                spanSM={null}
                spanXL={6}
              >
                <ReactMarkdown
                  astPlugins={Array []}
                  escapeHtml={true}
                  plugins={Array []}
                  rawSourcePos={false}
                  renderers={Object {}}
                  skipHtml={false}
                  source="Lorem ipsum dolor sit amet, consectetur adipisicing elit. Modi deserunt veritatis dolore sit ea, illum quae omnis placeat, natus est sapiente atque eaque obcaecati.

Lorem ipsum dolor sit amet, consectetur adipisicing elit. Modi deserunt veritatis dolore sit ea, illum quae omnis placeat, natus est sapiente atque eaque obcaecati."
                  sourcePos={false}
                  transformLinkUri={[Function]}
                />
              </Column>
              <Column
                pull={null}
                pullLG={null}
                pullMD={null}
                pullSM={null}
                pullXL={null}
                push={null}
                pushLG={null}
                pushMD={null}
                pushSM={null}
                pushXL={1}
                span={12}
                spanLG={null}
                spanMD={6}
                spanSM={null}
                spanXL={5}
              >
                <figure
                  className="portfolio-page__supporting-image"
                >
                  <img
                    alt=""
                    src="http://autotomic.com/envious/israelitish?a=rhonchal&b=tetrahydro#piehouse"
                  />
                </figure>
              </Column>
            </Row>
            <hr
              className="portfolio-page__content-divider"
            />
            <PortfolioCarousel
              desktopImage="http://hyperterrestrial.com/unconspiringness/uncreeping?a=pasterer&b=hemen#timalia"
              mobileImage="https://quadrupedan.com/inroader/ohmmeter?a=proseuche&b=provisorily#coriander"
              tabletImage="https://subtilism.com/dermatoscopy/nombril?a=phlebolithiasis&b=blancher#downthrow"
            />
            <PortfolioDeviceLineup
              desktopImage="http://hyperterrestrial.com/unconspiringness/uncreeping?a=pasterer&b=hemen#timalia"
              mobileImage="https://quadrupedan.com/inroader/ohmmeter?a=proseuche&b=provisorily#coriander"
              tabletImage="https://subtilism.com/dermatoscopy/nombril?a=phlebolithiasis&b=blancher#downthrow"
            />
            <hr
              className="portfolio-page__content-divider"
            />
            <div
              className="portfolio-page__external-link"
            >
              <Button
                external={true}
                primary={true}
                secondary={true}
                to="https://endocrinopathic.com/pluviometrically/phosphorylase?a=ulnocarpal&b=proreptilia#trochleariform"
              >
                Visit 
                Scrupulum Misphrase
              </Button>
            </div>
          </Column>
        </Row>
      </Container>
    </PortfolioPageContent>
  </div>
</withRouter(Page)>
`;
